ic-agent/src/identity/secp256k1.rs

```diff
@@ -87,7 +87,10 @@ impl Identity for Secp256k1Identity {
             .map_err(|err| format!("Cannot create secp256k1 signature: {}", err))?;
         let r = ecdsa_sig.r().as_ref().to_bytes();
         let s = ecdsa_sig.s().as_ref().to_bytes();
-        let mut bytes = [0; 64];
+        let mut bytes = [0u8; 64];
+        if r.len() > 32 || s.len() > 64 {
```

Comment:
This should be `s.len() > 32`

Reply:
🤦 thanks :)

Reply:
So `bytes` array should look like this?
```
|        rrr|      sssss|
0           32          64
```

Reply:
Yes exactly. Both `r` and `s` are (in this case) at most 32 bytes.